src/prpy/util.py

```diff
@@ -986,6 +1184,140 @@ def GetCollisionCheckPts(robot, traj, include_start=True, start_time=0.,
             dt = 2. * dt
 
 
+def GetLinearCollisionCheckPts(robot, traj, norm_order=2, sampling_order=None):
+    """
+    For a piece-wise linear trajectory, generate a list
+    of configuration pairs that need to be collision checked.
+
+    This will step along the trajectory from start to end
+    at a resolution that satisifies the specified error metric.
+
+    @param openravepy.Robot      robot: The robot.
+    @param openravepy.Trajectory traj:  The trajectory for which we need
+                                        to generate sample points.
+    @param int      norm_order: 1  ==>  The L1 norm
+                                2  ==>  The L2 norm
+                                inf  ==>  The L_infinity norm
+    @param string   sampling_order:
+                        'linear'         : Sample in a linear sequence
+                        'van_der_corput' : Sample in an optimal sequence
+
+    @returns generator: A tuple (t,q) of float values, being the sample
+                        time and joint configuration.
+    """
+
+    # If trajectory is already timed, strip the deltatime values
+    if IsTimedTrajectory(traj):
+        traj = UntimeTrajectory(traj)
+
+    traj_cspec = traj.GetConfigurationSpecification()
+
+    # Make sure trajectory is linear in joint space
+    try:
+        # type can be 'linear' or 'quadratic'
+        interp_type = traj_cspec.GetGroupFromName('joint_values').interpolation
+    except openravepy.openrave_exception:
+        raise ValueError('Failed calling GetGroupFromName()')
```

Comment:
A better error message would be something like "Trajectory does not have a `joint_values` group."